### PR TITLE
Custom reports updates

### PR DESCRIPTION
**Pull request by carkom**
Some slight UI and visual changes to custom reports

### CODE REVIEW

**Comment by Teprifer** [comment]
Confirming categories sort per budget order when importing my budget.

**Comment by Teprifer** [comment]
Ok this is interesting, thought I'd dig a little deeper as I originally just checked my regular deposits filter in the default and came across some odd behaviour now:

1. Click create new custom report
2. Click **Type** and _Net_ is disabled
3. Change **Split By** to _Account_
4. Change **Split By** back to _Category_
5. Click **Type** and _Net_ is now available for selection

image
_By way of explanation since I cut off categories, the far right is "Income" and the one in the middle is a category with Deposit and Payments categorised to it, net positive_


For comparison, this is the same graph, just changed **Type** back to _Payment_

image

**Comment by youngcw** [comment]
This is nice so far.  I think it would be better long term to try to not use so much screen space for the create new report button.  I would be a fan of a floating button, maybe at the bottom corner.

**Comment by MatissJanis** [comment]
> This is nice so far. I think it would be better long term to try to not use so much screen space for the create new report button. I would be a fan of a floating button, maybe at the bottom corner.

+1

IMO a better location for the button would be at the top of the page (screenshot). Plus we should make it smaller and `primary` color.

And then we can get rid of the separator line between the graphs. 

Screenshot 2024-02-28 at 08 01 57

**Comment by youngcw** [comment]
sankey report now takes up a whole row in the dash

**Comment by carkom** [comment]
> After this is there anything else you think NEEDS to be included? or is this about ready to remove the feature flag?

There's two things I'd like to finish before removing feature flag. 
1. incorporating "yearly" interval choice (I have a PR about 75% done)
2. create "disabledList" to make the functionality more consistent and the code easier to read (This PR is ready to submit)

My assumption is that the next release should have these PRs merged and a further PR to remove feature flag. Cheers!

**Comment by youngcw** [comment]
vrt needs fixed

**Comment by carkom** [comment]
> vrt needs fixed

Was just an errored out test, no VRT changes. Cheers!

**Review by MatissJanis — CHANGES_REQUESTED** [review]
The old widget size was visually much more pleasant. Can we revert this change back? There should not be any VRT screenshot updates in this PR. It should only update things to do with custom reports.

**Comment by MatissJanis** [comment]
> You mean the networth and cash flow cards?

Yes

**Comment by MatissJanis** [comment]
Screenshot 2024-02-29 at 19 41 12

Found one more bug here: the spacing between the header and the other content is too small.

**Comment by carkom** [comment]
Both fixed. Cheers!

File: packages/desktop-client/src/components/reports/reports/CustomReportListCards.tsx
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/reports/CustomReportListCards.tsx:92]: **🥜 nitpick:** ‏we can remove these lines to reduce the amount of duplication
```suggestion
```

**Review by MatissJanis — DISMISSED** [review]
Just a small nitpick on my end. Overall: LGTM!

(but please wait for another approval before merging as quite a few others were looking into this PR too)

File: packages/desktop-client/src/components/reports/Overview.jsx
Comment (twk3) [inline packages/desktop-client/src/components/reports/Overview.jsx:40]: This is currently navigating to the last loaded custom report rather than to a new custom report if used after looking at a saved report. I think we want it to open a default report each time?

File: packages/desktop-client/src/components/reports/ReportOptions.ts
Comment (twk3) [inline packages/desktop-client/src/components/reports/ReportOptions.ts:156]: suggestion: Can we comment the return to better describe the condition going on here. It's a pretty intense condition to just read later without context as to why.

File: packages/desktop-client/src/components/reports/reports/CustomReport.jsx
Comment (twk3) [inline packages/desktop-client/src/components/reports/reports/CustomReport.jsx:329]:
```suggestion
          {report.name || 'Unsaved report'}
```

Nullish operator isn't going to work here, because we have an empty string (falsey) rather than nullish.

**Review by twk3 — CHANGES_REQUESTED** [review]

File: packages/desktop-client/src/components/reports/reports/CustomReportListCards.tsx
Comment (carkom) [reply packages/desktop-client/src/components/reports/reports/CustomReportListCards.tsx:92]: This is not dupliacted. The state "true" keeps the menu open so the user can read the error. With this removed the error gets written to a menu that then immediately gets closed without time for the user to read it.

File: packages/desktop-client/src/components/reports/Overview.jsx
Comment (carkom) [reply packages/desktop-client/src/components/reports/Overview.jsx:40]: I'm not experiencing the same behavior. If I open a saved report then click "reports" in the main menu sidebar then click this button, I get a new default report. Is your experience different? How can I duplicate what you're seeing?

Comment (twk3) [reply packages/desktop-client/src/components/reports/Overview.jsx:40]: Looks like it starts happening if you load a custom report. Then go back to the cards and rename it from there. Then after the rename, click the button. (and it loads with the old name, prior to the rename)

Comment (carkom) [reply packages/desktop-client/src/components/reports/Overview.jsx]: I think I got it sorted, can you test again?

**Comment by Teprifer** [comment]
Appreciate adding the saved report name to the header, 2 verrry minor suggestions 'for another day' :) 

1) Remove the 's' in reports to make it singular as it's a single report being viewed.
2) Add the (modified) tag when doing so with right save/load menu.

e.g.

image

**Comment by carkom** [comment]
> Appreciate adding the saved report name to the header, 2 verrry minor suggestions 'for another day' :)
> 
>     1. Remove the 's' in reports to make it singular as it's a single report being viewed.
> 
>     2. Add the (modified) tag when doing so with right save/load menu.

I removed the "s" as that's an easy one. I've added the second suggestion to my feedback doc for a future PR.

**Review by twk3 — APPROVED** [review]
The changes look good. Thanks @carkom

**Pull request merged**
